lib/LaTeXML/Package/siunitx.sty.ltxml

```diff
@@ -402,156 +516,235 @@ sub six_groupdigits {
+      ($bracket > 1 ? six_get('close-bracket') : undef),
+      six_format_number($arg1, 1),
+      I_superscript({ operator_meaning => 'power' }, six_get('exponent-base'),
+        six_format_number($arg2))); }
+  # If mantissa is simple number, use scientific notation for the meaning
+  # (all the dual cruft above formats appropriately, but is wasted)
+  if ($arg1 && !$$arg1{operator} && (ToString(six_get('exponent-base')) eq '10')) {
+    $result = I_wrap({ meaning => T_OTHER(six_number_string($number)) }, $result); }
+  return $result; }
+
+sub six_format_compoundnumber {
+  my ($number, $bracket) = @_;
+  my $op   = $$number{operator};
+  my $arg1 = $$number{arg1};
+  my $arg2 = $$number{arg2};
+  my $result;
+  if ($op eq 'comparator') {
+    # NOTE: Semantic?
+    $result = I_wrap({},
+      ($$number{comparator} || ()),
+      six_format_number($arg1)); }
+  elsif ($op eq 'product') {
+    $result = six_format_infix(
+      six_get_op({ role => 'MULOP', meaning => 'times' }, 'output-product'), undef, undef,
+      six_format_number($arg1, 1),
+      six_format_number($arg2, 1)); }
+  elsif ($op eq 'quotient') {
+    if (six_getChoice('quotient-mode') eq 'fraction') {
+      $result = Tokens(
+        six_get('fraction-function'),    # Likely to be a macro! Hopefully has semantics?
+        T_BEGIN, six_format_number($arg1), T_END,
+        T_BEGIN, six_format_number($arg2), T_END); }
+    else {
+      $result = six_format_infix(
+        six_get_op({ role => 'MULOP', meaning => 'divide' }, 'output-quotient'), undef, undef,
+        six_format_number($arg1, 1),
+        six_format_number($arg2, 2)); } }
+  else {
+    Error('unexpected', $op, undef, "Unrecognized operator $op in siunitx number"); }
+  return $result; }
+
 sub six_format_number {
   my ($number, $bracket) = @_;
   return unless $number;
-  return $number unless ref $number eq 'HASH';
+  return I_wrap({}, $number) unless ref $number eq 'HASH';
   $bracket = 0 unless $bracket && six_getBool('bracket-numbers');
-  # Not ONLY format the number, but arrange for a fork representing the semantics!
   my @tokens = ();
   if (my $op = $$number{operator}) {
+    my $arg1 = $$number{arg1};
+    my $arg2 = $$number{arg2};
+    if ($op eq 'uncertain') { push(@tokens, six_format_uncertainnumber($number, $bracket)); }
+    elsif ($op eq 'complex') { push(@tokens, six_format_complexnumber($number, $bracket)); }
+    elsif ($op eq 'exponent') { push(@tokens, six_format_scinumber($number, $bracket)); }
+    else { push(@tokens, six_format_compoundnumber($number, $bracket)); } }
+  else {
+    push(@tokens, six_format_simplenumber($number, $bracket)); }
+  return Tokens(@tokens); }
+
+# Return the plain-text string for a number, for use in meaning attribute
+# Note that the format is pretty ad-hoc, except for simple-numbers
+# and scientific-notation, where the mantissa is a simple number.
+# You'll generally not want to use this for "meaning", except in those cases
+sub six_number_string {
```

Comment:
I wasn't going to say anything for a while, but `six_number` really got to me - same point as in physics.sty applies, we could always make a `Package::SIUnitx` namespace to host these methods. But if you'd really prefer a prefix, maybe something less confusing than `six`, such as `sx`?

Reply:
Yeah, this one's much worse than physics. I'd thought about experimenting with package namespaces, but put it off till I got the basic rewrite done.

Reply:
You can put si into its own package, but can't really import TeX (pool or package) without a lot of exports, right?  That sounds like a good idea, but probably best when factoring the pools, which should be deferred till next release.